pkg/client/auditor/auditor.go

```diff
@@ -380,6 +377,37 @@ func (a *defaultAuditor) audit() error {
 	return noErr
 }
 
+func (a *defaultAuditor) verifyStateSignature(
+	serverID string,
+	serverState *schema.ImmutableState,
+) error {
+
+	if a.serverSigningPubKey != nil && serverState.GetSignature() == nil {
+		return fmt.Errorf(
+			"a server signing public key has been specified for the auditor, "+
+				"but the state %s at TX %d received from server %s @ %s is not signed",
+			serverState.GetTxHash(), serverState.GetTxId(), serverID, a.serverAddress)
+	}
+
+	if serverState.GetSignature() != nil {
+		pk := a.serverSigningPubKey
+		if pk == nil {
+			x, y := elliptic.Unmarshal(elliptic.P256(), serverState.GetSignature().GetPublicKey())
```

Comment:
The code here is binded with elliptic p256 curve type. 
We should use an abstraction.
Inside Signer package there is an ecdsa implentation, we should modify Signer interface and expose UnmarshalKey method and then use it there.